readline/buffer.go

```diff
@@ -0,0 +1,301 @@
+package readline
+
+import (
+	"fmt"
+
+	"github.com/emirpasic/gods/lists/arraylist"
```

Comment:
Check out https://pkg.go.dev/container/list – might this be a good built-in alternative?

Reply:
It would work, but it doesn't have a `Get()` function to get a node at a specific index. You have to keep track of that yourself.

Reply:
👍

Reply:
Not important for merging, but would be curious if using `[]rune` could work– maybe with a custom type `type buf []rune` with some custom methods for the inserts/removes

Reply:
I do use `[]rune` when inserting lines into the history (which is now added). This is similar to how the original golang readline works. For the individual characters it uses a single rune as nodes in the linked list.

Back to the original point, I could write a convenience function to replace `Get()` which just iterates from the head to correct index. I'm assuming that's probably what is going on with the gods implementation. `Remove()` could work somewhat similarly. We can look at adding that in the future to cut out the dependency.